edx/analytics/tasks/reports/invoices.py

```diff
@@ -0,0 +1,273 @@
+            #     output_root=self.output_root,
+            #     date=day,
+            #     overwrite=self.overwrite,
+            # )
+
+    def output(self):
+        return [task.output() for task in self.requires()]
+
+
+class ShoppingCartInvoiceTransactionsByDayTask(InvoiceTransactionsTaskMixin, MapReduceJobTask):
+    """
+    Creates TSV file with information about ShoppingCart invoices.
+    """
+
+    def requires(self):
+        yield (
+            ShoppingCartInvoiceTransactionsTask(
+                import_date=self.date
+            )
+        )
+
+    def mapper(self, line):
+        transaction_id, date_created, amount, currency, status, invoice_id = line.split('\t')
+        transaction_type = 'sale' if amount >= 0 else 'refund'
+        transaction_record = [
+            date_created,
+            DEFAULT_PAYMENT_GATEWAY_ID,
+            DEFAULT_PAYMENT_GATEWAY_ACCOUNT_ID,
+            invoice_id,
+            currency,
+            amount,
+            DEFAULT_TRANSACTION_FEE,
+            transaction_type,
+            DEFAULT_PAYMENT_METHOD,
+            DEFAULT_PAYMENT_METHOD_TYPE,
+            transaction_id,
+        ]
+        yield transaction_id, transaction_record
+
+    def reducer(self, _key, values):
+        for transaction_record in values:
+            with self.output().open('w') as output_tsv_file:
+                output_tsv_file.write('\t'.join(transaction_record) + '\n')
+
+    def output(self):
+        return get_target_from_url(
+            url_path_join(self.output_root, 'payments', 'dt=' + self.date.isoformat(), 'shoppingcart_invoices.tsv')
+        )
+
+
+class OttoInvoiceTransactionsByDayTask(InvoiceTransactionsTaskMixin, MapReduceJobTask):
+    """
+    Creates TSV file with information about Otto invoices.
+    """
+
+    def requires(self):
+        yield (
+            OttoInvoiceTransactionsTask(
+                import_date=self.date
+            )
+        )
+
+    def mapper(self, line):
+        invoice_id, date_created, state, order_id, currency, amount = line.split('\t')
+        transaction_type = 'sale' if amount >= 0 else 'refund'
+        invoice_record = [
+            date_created,
+            DEFAULT_PAYMENT_GATEWAY_ID,
+            DEFAULT_PAYMENT_GATEWAY_ACCOUNT_ID,
+            order_id,
+            currency,
+            amount,
+            DEFAULT_TRANSACTION_FEE,
+            transaction_type,
+            DEFAULT_PAYMENT_METHOD,
+            DEFAULT_PAYMENT_METHOD_TYPE,
+            invoice_id,
+        ]
+        yield invoice_id, invoice_record
+
+    def reducer(self, _key, values):
+        for invoice_record in values:
+            with self.output().open('w') as output_tsv_file:
+                output_tsv_file.write('\t'.join(invoice_record) + '\n')
+
+    def output(self):
+        return get_target_from_url(
+            url_path_join(self.output_root, 'payments', 'dt=' + self.date.isoformat(), 'otto_invoices.tsv')
+        )
+
```

Comment:
These map-reduce jobs are rather simple, so I wonder whether it would be easier/more efficient to implement them as Hive SQL queries instead.  I don't have experience with the performance of these jobs, so this is just an educated guess.  We might ask upstream for their opinion.